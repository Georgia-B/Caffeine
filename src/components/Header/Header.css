.header {
    display: flex;
    align-items: center;
    margin: 0 20px;
}

.header__title {
    flex: 2;
    text-align: center;
}

.header__action {
    flex: 1;
    text-align: right;
}

.header__spacer {
    flex: 1;
}