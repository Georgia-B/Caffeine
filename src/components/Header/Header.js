import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Button from '../Button/Button';
import styles from './Header.module.css';

const Header = ({ title, action }) => {
    return (
        <div className={styles.header}>
            {action && <div className={styles.spacer}></div>}
            <h1 className={styles.title}>{title}</h1>
            {action &&
                <div className={styles.action}>
                    <Link to={action.to}>
                        <Button text={action.label} />
                    </Link>
                </div>
            }
        </div>
    );
};

Header.propTypes = {
    title: PropTypes.string,
    action: PropTypes.object,
};

export default Header;
