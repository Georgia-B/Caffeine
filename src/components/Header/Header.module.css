.header {
    display: flex;
    align-items: center;
    margin: 0 20px;
}

.title {
    flex: 2;
    text-align: center;
}

.action {
    flex: 1;
    text-align: right;
}

.spacer {
    flex: 1;
}

.link:focus {
    outline: none;
}