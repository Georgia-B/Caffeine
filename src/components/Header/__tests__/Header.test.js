import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { renderRouter } from '../../../test';
import Header from '../Header';

describe('Components | <Header />', () => {
    afterEach(cleanup);
    it('should render correctly without action', () => {
        const component = render(<Header
            title="Hello World"
        />);

        expect(component.asFragment()).toMatchSnapshot();
    });

    it('should render correctly with action', () => {
        const component = renderRouter(
            <Header
                title="Hello World"
                action={{ to: '/', label: 'Click me' }}
            />);

        expect(component.asFragment()).toMatchSnapshot();
    });
});
