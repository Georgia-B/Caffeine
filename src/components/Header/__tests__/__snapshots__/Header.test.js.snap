// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | <Header /> should render correctly with action 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="spacer"
    />
    <h1
      class="title"
    >
      Hello World
    </h1>
    <div
      class="action"
    >
      <a
        class="link"
        href="/"
      >
        <button
          class="button"
        >
          Click me
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Components | <Header /> should render correctly without action 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <h1
      class="title"
    >
      Hello World
    </h1>
  </div>
</DocumentFragment>
`;
